kotlin-generator/src/kt-client.ts

```diff
@@ -0,0 +1,215 @@
+import { AstRoot } from "@sdkgen/parser";
+import { generateClass, generateEnum, generateErrorClass, generateTypeName, generateJsonAddRepresentation, mangle } from "./helpers";
+
+interface Options {}
+
+export function generateKtClientSource(ast: AstRoot, options: Options) {
+    let code = "";
+
+    code += `
+@file:Suppress("UnusedImport")
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.pm.PackageManager
+import android.graphics.Point
+import android.os.Build
+import android.provider.Settings
+import android.util.Base64
+import android.util.Log
+import android.view.WindowManager
+import com.google.gson.*
+import com.google.gson.reflect.TypeToken
+import io.sdkgen.runtime.SdkGenHttpClient
+import java.io.IOException
+import java.io.InvalidObjectException
+import java.io.Serializable
+import java.lang.reflect.Type
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import java.util.*
+import java.util.concurrent.TimeUnit
+import kotlin.concurrent.timerTask
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+import kotlin.experimental.and
+import kotlinx.coroutines.*
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOn
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+
+inline fun <reified T> Gson.fromJson(json: String) = fromJson<T>(json, object: TypeToken<T>() {}.type)
+inline fun <reified T> Gson.fromJson(json: JsonElement) = fromJson<T>(json, object: TypeToken<T>() {}.type)
+
+@ExperimentalCoroutinesApi
+@Suppress("DeferredIsResult", "unused")
+@SuppressLint("SimpleDateFormat", "StaticFieldLeak")
+object API { \n`;
```

Comment:
This file should expose a class that inherits from SdkgenHttpClient. The application is supposed to create an instance of this class and pass it along. It allows multiple instances to exist and helpes with dependency injection. It shouldn't be a global singleton.

Reply:
This makes sense, but if we are using this logic, we should separete the models enums and interfaces from this class.

Reply:
They can be static innerclasses